client/client-multi/client-objective-c/Kaa/channel/impl/channels/DefaultOperationTcpChannel.m

```diff
@@ -650,26 +650,28 @@ - (instancetype)initWithChannel:(DefaultOperationTcpChannel *)channel {
 }
 
 - (void)main {
-    [NSThread sleepForTimeInterval:PING_TIMEOUT_SEC];
-    
-    if (self.isCancelled || self.isFinished) {
-        DDLogInfo(@"%@ Can't execute ping task for channel [%@]. Task was cancelled.", TAG, [self.channel getId]);
-        return;
-    }
-    
-    @try {
-        DDLogInfo(@"%@ Executing ping task for channel [%@]", TAG, [self.channel getId]);
-        [self.channel sendPingRequest];
-        if (self.isCancelled) {
-            DDLogInfo(@"%@ Can't schedule new ping task for channel [%@]. Task was cancelled.", TAG, [self.channel getId]);
-        } else {
-            [self.channel schedulePingTask];
+    dispatch_time_t time =  dispatch_time(DISPATCH_TIME_NOW, (int64_t)PING_TIMEOUT_SEC * NSEC_PER_SEC);
+    dispatch_queue_t queue =  dispatch_queue_create("com.kaaproject.pingtask", NULL);
```

Comment:
Better to take some global queue with low priority(QOS_CLASS_UTILITY probably fits best) than create it manually every time.